homeassistant/components/shopping_list/__init__.py

```diff
@@ -281,3 +320,23 @@ def websocket_handle_clear(hass, connection, msg):
     hass.data[DOMAIN].async_clear_completed()
     hass.bus.async_fire(EVENT, {"action": "clear"})
     connection.send_message(websocket_api.result_message(msg["id"]))
+
+
+@callback
+def websocket_handle_reorder(hass, connection, msg):
+    """Handle reordering shopping_list items."""
+    msg_id = msg.pop("id")
+    try:
+        hass.data[DOMAIN].async_reorder(msg.pop("item_ids"))
+        hass.bus.async_fire(EVENT, {"action": "reorder"})
+        connection.send_message(websocket_api.result_message(msg_id))
+    except KeyError:
+        connection.send_message(
+            websocket_api.error_message(
+                msg_id, "item_not_found", "One or more item id(s) not found."
+            )
+        )
+    except vol.Invalid as err:
+        connection.send_message(
+            websocket_api.error_message(msg_id, "bad_request", f"{err}")
```

Comment:
Use the websocket_api error `ERR_INVALID_FORMAT`

Reply:
fixed